server/tabby_server/vision/yolo_test.py

```diff
@@ -16,53 +17,104 @@
 
 # Instantiates the model.
 # Runs it. Outputs files to vision/example_yolo.
+# https://docs.ultralytics.com/modes/predict/#working-with-results
 
 
-def ultralytics_shelf_detection(file_path, save_output):
+def ultralytics_shelf_detection(file_path) -> dict[str, list]:
     # Load pretrained model.
     model = YOLO(file_path + "shelf_yolo.pt")
 
+    # ai-gen start (ChatGPT-4.0, 1)
+
+    import cv2
+
+    # Reads image as a NumPy array
+    image = cv2.imread(file_path + "example_shelves/shelf_2.jpg")
+    image_rgb = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+    # Resize to model's expected input
+    resized_image = cv2.resize(image_rgb, (640, 640))
+    # Normalize pixel values
+    normalized_image = resized_image / 255.0
+    import torch
+
+    tensor_image = (
+        torch.from_numpy(normalized_image)
+        .float()
+        .permute(2, 0, 1)
+        .unsqueeze(0)
+    )
+
+    # ai-gen end
```

Comment:
Here is the ChatGPT conversation that made this: https://chatgpt.com/share/67359936-d608-800f-a86e-3eab0a9e9340